spec/spec.tex

```diff
@@ -1024,6 +1024,45 @@ \subsubsection{Format Strings}
 \item \verb|\'| : Single quote
 \end{itemize}
 
+\subsection{Verification}
+
+To facilitate simulation, model checking and formal methods, there are three non-synthesizable verification statements available: assert, assume and cover. Backends are free to generate the corresponding model checking constructs in the target language, but this is not required by the FIRRTL specification. For example, the System Verilog emitter produces System Verilog assert, assume and cover statements. However, backends should not generate any new synthesizable hardware when handling to verification statements. Backends that do not obey this rule should issue a warning.
+
+Each type of verification statement requires a clock signal, a predicate signal, an enable signal and an explanatory message string literal. The predicate and enable signals must have single bit unsigned integer type. When an assert is violated the explanatory message may be issued as guidance. The explanatory message may be phrased as if prefixed by the words ``Verifies that...''.
+
+\subsubsection{Assert}
+
+The assert statement verifies that the predicate is true on the rising edge of any clock cycle when the enable is true. In other words, it verifies that enable implies predicate.
+
+\begin{lstlisting}
+wire clk:Clock
+wire pred:UInt<1>
```

Comment:
how about we change that to `node pred = eq(X, Y)`?

Reply:
Do you think it's important to explicitly call out that the type of pred is `UInt<1>`?

Reply:
That is the thing that gets lost in my suggestion. Yeah, maybe we should have a `wire` and a `connect` which would combine both points.

```
wire pred: UInt<1>
pred <= eq(X, Y)
```